packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -170,6 +196,95 @@ function areHookInputsEqual(
   return true;
 }
 
+// till we have String::padEnd, a small function to
+// right-pad strings with spaces till a minimum length
+function padEndSpaces(string: string, length: number) {
+  if (__DEV__) {
+    if (string.length >= length) {
+      return string;
+    } else {
+      return string + ' ' + new Array(length - string.length).join(' ');
+    }
+  }
+}
+
+function flushHookMismatchWarnings() {
+  // we'll show the diff of the low level hooks,
+  // and a stack trace so the dev can locate where
+  // the first mismatch is coming from
+  if (__DEV__) {
+    if (currentHookMismatch) {
```

Comment:
Just noticed this: for future reference, we try to always explicitly compare to null or undefined. That way we distinguish between intentional and unintentional missing values, and comparisons are slightly faster.